index.js

```diff
@@ -18,5 +19,17 @@ module.exports = (red, green, blue) => {
 		throw new TypeError('Expected three numbers below 256');
 	}
 
-	return ((blue | green << 8 | red << 16) | 1 << 24).toString(16).slice(1);
+	if (typeof alpha === 'number') {
+		if (alpha < 1 && alpha >= 0) {
+			alpha = Math.round(256 * alpha).toString(16);
+		} else if (alpha > 1 && alpha < 100) {
+			alpha = Math.round(256 * alpha / 100).toString(16);
+		} else {
+			throw new TypeError('Expected alpha value as a decimal or percentage');
+		}
+	} else {
+		alpha = '';
+	}
+
+	return ((blue | green << 8 | red << 16) | 1 << 24).toString(16).slice(1) + alpha;
```

Comment:
I would calculate this first, then determine if alpha is there and if so shift this 8 bytes and add alpha.